### PR TITLE
exposed FP16Optimizer and added documentation

### PR DESCRIPTION
**Pull request by AdamLouly**
Exposing FP16 Optimizer to torch-ort.
Adding documentation of how to use it.
Re Adding deleted FusedAdam and DataBalancer Documentation

### CODE REVIEW

**Review by askhade — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -117,6 +117,71 @@ python -m build
...
 The torch-ort library accelerates training of large transformer PyTorch models to reduce the training time and GPU cost with a few lines of code change. It is built on top of highly successful and proven technologies of ONNX Runtime and ONNX format and includes the ONNX Runtime Optimizer and Data Sampler.
+## Add ONNX Runtime for PyTorch to your PyTorch training script
+
+```python
+from torch_ort import ORTModule
+model = ORTModule(model)
+# PyTorch training script follows
+```
+
+## Usage of FusedAdam and FP16 Optimizer (Optional)
```
Comment (baijumeswani) [inline README.md:128]: Why optional?

File: README.md [hunk 2]
```diff
@@ -117,6 +117,71 @@ python -m build
...
+
```
Comment (baijumeswani) [inline README.md:129]: Shall we give an intro on what FP16_Optimizer is?

File: README.md (hunk 1 above)
Comment (AdamLouly) [reply README.md:128]: like its not required to be used in this example, only if you want to use fp16 you can add that line.

File: README.md (hunk 2 above)
Comment (AdamLouly) [reply README.md:129]: I think we can add a link to fp16_optimizer.py like FusedAdam.

File: README.md (hunk 1 above)
Comment (baijumeswani) [reply README.md:128]: I don't think we need to provide the optional keyword. I think with some details we should be ok.

File: README.md (hunk 2 above)
Comment (baijumeswani) [reply README.md:129]: if possible, let's add a line or two introducing what it does and then ask users to read FP16_Optimizer.py for more details.

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**